File: Desafio Mobile iOS/Desafio Mobile iOS/Models/Repository.m
//
//  Repository.m
//  Desafio Mobile iOS
//
//  Created by Adriano Rezena on 10/10/17.
//  Copyright © 2017 Adriano Rezena. All rights reserved.
//

#import "Repository.h"

@implementation Repository

-(id)initWithDictionary:(NSDictionary *)dictionary {
    if (self = [super init]) {
        self.id = [dictionary[@"id"] integerValue];
        self.name = dictionary[@"name"];
        self.repDescription = dictionary[@"description"];
        self.forks_count = [dictionary[@"forks_count"] integerValue];
        self.stargazers_count = [dictionary[@"stargazers_count"] integerValue];
        self.pulls_url = [dictionary[@"pulls_url"] stringByReplacingOccurrencesOfString:@"{/number}" withString:@""];
        
        self.owner = [[Owner alloc] initWithDictionary:dictionary[@"owner"]];
    }

    return self;
}


+ (NSString *)primaryKey {
    return @"id";
}


@end

File: Git Java/Git Java/Model/Repository.swift
//
//  Repository.swift
//  Git Java
//
//  Created by Filipe Amaral Neis on 25/07/2018.
//  Copyright © 2018 Neis. All rights reserved.
//

import Foundation

struct Repository: Decodable {
    var idRepository: Int?
    var name: String?
    var description: String?
    var forkCount: Int?
    var starCount: Int?
    var owner: Owner?
    
    private enum CodingKeys: String, CodingKey {
        case idRepository = "id",
        name, description, owner,
        forkCount = "forks_count",
        starCount = "stargazers_count"
    }
}
